src/coap_blockwise.c

```diff
@@ -251,7 +245,13 @@ enum golioth_status golioth_blockwise_post(struct golioth_client *client,
 
     if (set_cb)
     {
-        set_cb(client, &ctx->response, path, callback_arg);
+        bool local_error =
+            (status != GOLIOTH_OK && (ctx->block_idx == 0 || ctx->response.status == GOLIOTH_OK));
```

Comment:
I'm a little worried about this being brittle, since it's not obvious that the response from the previous block should remain useable while processing the next block. I think just adding a comment explaining this boolean logic should suffice for now.